ui/e2e/explorer/workflow/settings.spec.ts

```diff
@@ -193,3 +198,99 @@ test("it is possible to delete variables", async ({ page }) => {
     "the variable is no longer rendered in the list"
   ).toHaveCount(variables.length - 1);
 });
+
+test("it is not possible to create a variable with a name that already exists", async ({
+  page,
+}) => {
+  /* set up test data */
+  const variables = await createWorkflowVariables(namespace, workflow, 4);
+  const reservedName = variables[0]?.data.name ?? "";
+
+  await page.goto(`/${namespace}/explorer/workflow/settings/${workflow}`);
+
+  await page.getByTestId("variable-create").click();
+
+  page.getByTestId("variable-name").fill(reservedName);
+
+  await page.getByRole("button", { name: "Create" }).click();
+
+  await expect(
+    page.getByText("The name already exists"),
+    "it renders an error message"
+  ).toBeVisible();
```

Comment:
Good idea. The problem here is, that the button is not disabled at this point, that would be my preferred way to back up the test. 

I could add an event listener, but it always feels a bit counterintuitive to add a check that makes sure things are NOT happening. If the path to the endpoint would change in the future, this would not make the test fail, but silently lose the coverage of it.

One solution could be to add an API request to check if the resource has changed, but if you are okay with it, I would be fine to leave this test as it is and only test for the error message. Otherwise, I would feel the need to back up a lot of other test this way as well (like e.g. renaming files and folders, gateway and service forms, etc...)

Reply:
I think that's ok and you're probably right, there are probably lots of tests with the same limitation.

Considering the way our validation generally works, it is also pretty unlikely that a request is sent when a validation error occurs.